project.clj

```diff
@@ -50,6 +51,14 @@
                               "phantom-advanced" ["phantomjs"
                                                   "src/test/js/runner.js"
                                                   "src/test/html/advanced.html"]}}
+  
+  ;; Factoring out the duplication of this test selector function causes an error,
```

Comment:
I don't care if you copy/paste since it is minimal and making it general is probably more ceremony than needed.

Leiningen has first-class for arbitrary evaluation to create dynamic project.cjl's like this though.  It is used extensively in real-world lein projects as well as lein documentation sample project that most things can be found in.
e.g.
https://github.com/technomancy/leiningen/blob/stable/sample.project.clj#L420

It is used all over the place though.  lein doesn't always do a great job of documenting every feature.  I think the expectation is to explore more and even read the codebase function documentation for some things.  There are also some StackOverflow answers that use approaches to lein problems with `~`.  

See 
https://github.com/technomancy/leiningen/blob/stable/sample.project.clj#L420
for where the main doc is for this.